src/serializer/functions.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { BabelNode, BabelNodeCallExpression, BabelNodeIdentifier } from "babel-types";
+import { CompilerDiagnostic, FatalError } from "../errors.js";
+import invariant from "../invariant.js";
+import { intersectEffects } from "../methods/join.js";
+import { ExecutionContext, type PropertyBindings, Realm } from "../realm.js";
+import type { PropertyBinding } from "../types.js";
+import { EmptyValue, FunctionValue, Value } from "../values/index.js";
+import * as t from "babel-types";
+
+export class Functions {
+  constructor(realm: Realm, functions: ?Array<string>) {
+    this.realm = realm;
+    this.functions = functions;
+  }
+
+  realm: Realm;
+  functions: ?Array<string>;
+
+  evaluateNodeForEffects(node: BabelNode) {
```

Comment:
This seems like a generally useful function. I can see how we can cut a few lines e.g. in modules.js by using this.